DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/DateShiftProcessor.cs

```diff
@@ -0,0 +1,86 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Microsoft.Health.Dicom.DeID.SharedLib.Settings;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class DateShiftProcessor : IAnonymizerProcessor
+    {
+        private readonly DateShiftFunction _dateShiftFunction;
+        private readonly DateShiftScope _dateShiftScope = DateShiftScope.SopInstance;
+
+        public DateShiftProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
+            var dateShiftSetting = settingFactory.CreateAnonymizerSetting<DateShiftSetting>(settingObject);
+            _dateShiftFunction = new DateShiftFunction(dateShiftSetting);
+            if (settingObject.TryGetValue("DateShiftScope", StringComparison.OrdinalIgnoreCase, out JToken scope))
+            {
+                _dateShiftScope = (DateShiftScope)Enum.Parse(typeof(DateShiftScope), scope.ToString());
+            }
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+            EnsureArg.IsNotNull(context, nameof(context));
+
+            _dateShiftFunction.DateShiftKeyPrefix = _dateShiftScope switch
+            {
+                DateShiftScope.StudyInstance => context.StudyInstanceUID ?? string.Empty,
+                DateShiftScope.SeriesInstance => context.StudyInstanceUID ?? string.Empty,
+                DateShiftScope.SopInstance => context.SopInstanceUID ?? string.Empty,
+                _ => string.Empty,
+            };
+            if (item.ValueRepresentation == DicomVR.DA)
+            {
+                var values = Utility.ParseDicomDate((DicomDate)item).Select(_dateShiftFunction.ShiftDateTime).Where(x => !DateTimeUtility.IndicateAgeOverThreshold(x));
+                dicomDataset.AddOrUpdate(item.ValueRepresentation, item.Tag, values.Select(Utility.GenerateDicomDateString).ToArray());
+            }
+            else if (item.ValueRepresentation == DicomVR.DT)
```

Comment:
the logic is actually the same as that for DicomVR.DA.
The only difference is underlying data type.
Make an overload version of DateShiftFunction::ShiftDateTime(DateTimeObject) and IndicateAgeOverThreshold(DateTimeObject), you only need one piece of code here.